console-subscriber/src/aggregator/mod.rs

```diff
@@ -689,63 +703,61 @@ impl Aggregator {
                 );
             }
 
-            Event::ResourceOp {
+            Event::PollOp {
                 metadata,
                 at,
                 resource_id,
                 op_name,
-                op_type,
-                ..
+                async_op_id,
+                task_id,
+                readiness,
             } => {
-                match op_type.clone() {
-                    OpType::StateUpdate(attr_updates) => {
-                        if let Some(mut stats) = self.resource_stats.update(&resource_id) {
-                            for upd in attr_updates {
-                                match stats.attributes.get_mut(&upd.name) {
-                                    Some(attr) => attr.update(&upd.val),
-                                    None => {
-                                        stats.attributes.insert(upd.name.clone(), upd.val.into());
-                                    }
-                                }
-                            }
-                        }
-                    }
-                    OpType::Poll {
-                        async_op_id,
-                        task_id,
-                        readiness,
-                    } => {
-                        // TODO: make it less hacky. Ideally we want to store
-                        // these in a vec that gets cleaned periodically and allows
-                        // us to send only the new ones much like the IdData map.
-                        let mut async_op_stats = self.async_op_stats.update_or_default(async_op_id);
-                        async_op_stats.poll_stats.polls += 1;
-                        async_op_stats.latest_poll_op = Some(metadata.into());
-                        if async_op_stats.task_id.is_none() {
-                            async_op_stats.task_id = Some(task_id);
-                        }
-                        if async_op_stats.resource_id.is_none() {
-                            async_op_stats.resource_id = Some(resource_id.clone());
-                        }
+                let async_op_id = self.task_ids.id_for(async_op_id);
+                let resource_id = self.task_ids.id_for(resource_id);
+                let task_id = self.task_ids.id_for(task_id);
+
+                let mut async_op_stats = self.async_op_stats.update_or_default(async_op_id);
+                async_op_stats.poll_stats.polls += 1;
+                async_op_stats.task_id.get_or_insert(task_id);
+                async_op_stats.resource_id.get_or_insert(resource_id);
+
+                if matches!(readiness, Readiness::Pending)
```

Comment:
nit/TIOLI: if `Readiness` derived `Eq` and `PartialEq`, this could just be
```suggestion
                if readiness == Readiness::Pending
```